### DISCUSSION TITLE
Normal Buffer of scene

### DISCUSSION THREAD

**Discussion by KallkaGo** [pmndrs/postprocessing#662] [Q&A] [answered]
I tried using scene.overrideMaterial to replace the material in the scene with MeshNormalMaterial and render it to the rendertarget so that my custom post-processing can pick it up, but I found that when I sampled in the shader, the colors were different, even though I had set the renderer's outputColorSpace and the texture's colorSpace

Rendering directly to the screen：
![QQ_1728364550166](https://github.com/user-attachments/assets/3996b068-80e3-42c5-9729-baa1b49ca52c)

Sampled in custom effect：
![QQ_1728364634128](https://github.com/user-attachments/assets/9cf4bf13-87ed-4580-a492-a7fc48a76dda)

Am I overlooking something? Do i need gamma correction?

**Comment by vanruesc** [comment, marked as answer]
Depending on what passes you're using, you may need to set `effectPass.fullscreenMaterial.encodeOutput` to `false`.

**Reply by vanruesc** [reply]
In v7 it's `effectPass.fullscreenMaterial.colorSpaceConversion`.

**Reply by KallkaGo** [reply]
Thanks for your reply，during this period, I checked the source code and supplemented the relevant knowledge of Color Management. Now I have figured out the reason for this difference.
